crates/iota-core/src/execution_cache/unit_tests/writeback_cache_tests.rs

```diff
@@ -1328,6 +1328,62 @@ async fn latest_object_cache_race_test() {
     invalidator.join().unwrap();
 }
 
+#[tokio::test]
+async fn test_transaction_cache_race() {
+    telemetry_subscribers::init_for_testing();
+    let mut s = Scenario::new(None, Arc::new(AtomicU32::new(0))).await;
+    let cache = s.cache.clone();
+    let mut txns = Vec::new();
+
+    for i in 0..1000 {
+        let a = i * 4;
+        s.with_created(&[a]);
+        s.do_tx().await;
+
+        let outputs = s.take_outputs();
+        let tx = (*outputs.transaction).clone();
+        let effects = outputs.effects.clone();
+
+        txns.push((tx, effects));
+    }
+
+    let barrier = Arc::new(std::sync::Barrier::new(2));
```

Comment:
`Arc` is not needed here.

Reply:
Why is it not needed? It's used in different threads.